common/src/main/java/com/wynntils/mc/mixin/KeyMappingMixin.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.mc.mixin;
+
+import com.mojang.blaze3d.platform.InputConstants;
+import java.util.Map;
+import net.minecraft.client.KeyMapping;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(KeyMapping.class)
+public class KeyMappingMixin {
+    @Shadow @Final private static Map<String, Integer> CATEGORY_SORT_ORDER;
+
+    @Inject(
+            method =
+                    "<init>(Ljava/lang/String;Lcom/mojang/blaze3d/platform/InputConstants$Type;ILjava/lang/String;)V",
+            at = @At("RETURN"))
+    public void initPost(
+            String name, InputConstants.Type type, int i, String category, CallbackInfo ci) {
+        if (CATEGORY_SORT_ORDER.containsKey(category)) return;
```

Comment:
@Incompleteusern Sorry for not reviewing this earlier. Can you extract this code in the external class like other mixins, and just pass the CATEGORY_SORT_ORDER as argument? Code in mixins is hard to debug. You can't put breakpoints or do hot swapping, so the idea is to make the actual mixin as minimal as possible, just a hook.

Reply:
This code should be relatively invariable so it is likely fine to stay as is, furthermore, I don't see this changing in the future much

Reply:
Feel free to extract it though